dracut/anaconda-lib.sh

```diff
@@ -434,3 +434,14 @@ wait_for_updates() {
 wait_for_dd() {
     echo "[ -e /tmp/dd.done ]" > "$hookdir/initqueue/finished/dd.sh"
 }
+
+wait_for_disks() {
+    # Allow up to 'inst.wait_for_disks' seconds for disks to be enumerated and
+    # related udev rules to execute (defaults to 5 seconds, 0 disables the
+    # feature)
+    finished_hook="$hookdir/initqueue/finished/wait_for_disks.sh"
+    [ ! -e "$finished_hook" ] || return
+    DISKS_WAIT_DELAY=$(getargnum 5 0 10000 inst.wait_for_disks)
+    timeout=$(($(awk '/^now/ {print substr($3, 1, length($3) - 9); exit}' /proc/timer_list) + $DISKS_WAIT_DELAY))
+    echo "[ \$(awk '/^now/ {print substr(\$3, 1, length(\$3) - 9); exit}' /proc/timer_list) -ge $timeout ]" > $finished_hook
```

Comment:
That is a really fancy way how to wait 5 seconds :-D Any reason why not just use /proc/uptime?
Also, in theory, since all the hooks are sourced you can share variables and you could do sleep 1; $((CNT++))

But this will definitely work, it is just hard to understand what the code does.

Reply:
I will use `/proc/uptime`, but I don't want to sleep.